packages/core/src/bookmark-manager.ts

```diff
@@ -119,69 +91,36 @@ export function bookmarkManager (config: BookmarkManagerConfig = {}): BookmarkMa
 
 class Neo4jBookmarkManager implements BookmarkManager {
   constructor (
-    private readonly _bookmarksPerDb: Map<string, Set<string>>,
-    private readonly _bookmarksSupplier?: (database?: string) => Promise<Iterable<string>>,
-    private readonly _bookmarksConsumer?: (database: string, bookmark: Iterable<string>) => Promise<void>
+    private readonly _bookmarks: Set<string>,
+    private readonly _bookmarksSupplier?: () => Promise<Iterable<string>>,
+    private readonly _bookmarksConsumer?: (bookmark: Iterable<string>) => Promise<void>
   ) {
 
   }
 
-  async updateBookmarks (database: string, previousBookmarks: Iterable<string>, newBookmarks: Iterable<string>): Promise<void> {
-    const bookmarks = this._getOrInitializeBookmarks(database)
+  async updateBookmarks (previousBookmarks: Iterable<string>, newBookmarks: Iterable<string>): Promise<void> {
+    const bookmarks = this._bookmarks
     for (const bm of previousBookmarks) {
       bookmarks.delete(bm)
     }
     for (const bm of newBookmarks) {
       bookmarks.add(bm)
     }
     if (typeof this._bookmarksConsumer === 'function') {
```

Comment:
this check could be moved to constructor, if it's not a function raise an error. avoids needing to validate the type each call.

Reply:
But I think it can be worthy to add some validation in factory. I will do it a next PR.